### PR TITLE
Divider component

### PR DESCRIPTION
**Pull request by crstnmac**
SCR-20221104-iac

https://user-images.githubusercontent.com/20360870/199918640-827b3766-161c-45d4-a0e5-ac69fcc42bca.mov

### CODE REVIEW

File: src/components/divider/Divider.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        {label ? (
+          <Button themeColor="base" variant="outline" {...buttonProps}>
+            {label}
+          </Button>
+        ) : slot ? (
+          slot
+        ) : null}
+      </Box>
+
+      <Box style={tailwind.style(cx(dividerTheme.lines[orientation]))} />
```
Comment (Karthik-B-06) [inline src/components/divider/Divider.tsx:47]: I think we can remove this <Box />, as the slot is absolutely positioned. The Divider would work fine.

**Review by Karthik-B-06 — COMMENTED** [review]
Check my comments.

File: src/components/divider/Divider.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  const tailwind = useTheme();
+  const dividerTheme = useTheme("divider");
+  return (
+    <Box
+      style={tailwind.style(
+        cx(dividerTheme.orientation[orientation]),
+        orientation === "vertical" ? "overflow-hidden" : {},
+      )}
+    >
+      <Box style={tailwind.style(cx(dividerTheme.lines[orientation]))} />
```
Comment (Karthik-B-06) [inline src/components/divider/Divider.tsx:31]: I think there should be a way to style the line, to handle one of the cases users might have.

File: src/components/divider/Divider.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { ReactElement } from "react";
+import { Box, useTheme } from "@adaptui/react-native-tailwind";
+
+import { createComponent, cx } from "../../utils";
+import { Button, ButtonProps } from "../button";
+
+interface DividerProps {
+  label?: string;
```
[the remark this answers is not in the document]
Comment (crstnmac) [reply src/components/divider/Divider.tsx:8]: Fixed

File: src/components/divider/Divider.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  orientation?: "horizontal" | "vertical";
+  slot?: ReactElement;
+  labelPosition: "start" | "center" | "end";
```
[the remark this answers is not in the document]
Comment (crstnmac) [reply src/components/divider/Divider.tsx:11]: Added descriptions to all props

File: src/theme/defaultTheme/divider.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+export const divider = {
```
[the remark this answers is not in the document]
Comment (crstnmac) [reply src/theme/defaultTheme/divider.ts:1]: used this style to define the object

File: src/components/divider/Divider.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  buttonProps: ButtonProps;
+}
+
+const RNDivider = ({
+  label,
```
[the remark this answers is not in the document]
Comment (crstnmac) [reply src/components/divider/Divider.tsx:16]: Moved it

File: src/components/divider/Divider.tsx (hunk 2 above)
Comment (crstnmac) [reply src/components/divider/Divider.tsx:31]: Added an additional prop to style the divider line

**Review by Karthik-B-06 — APPROVED** [review]
LGTM 🎉

**Pull request merged**